kpi/tests/api/v2/test_api_submissions.py

```diff
@@ -599,6 +599,32 @@ def test_list_submissions_asset_publicly_shared_and_shared_with_user_as_anotheru
         self.asset.remove_perm(anonymous_user, PERM_VIEW_ASSET)
         self.asset.remove_perm(anonymous_user, PERM_VIEW_SUBMISSIONS)
 
+    def test_list_query_elem_match(self):
+        """
+        Ensure query is able to filter on an array
+        """
+        submission = self.submissions[0]
+        group = 'group_lx4sf58'
+        question = 'q3'
+        submission[group] = [
+            {
+                f'{group}/{question}': 'whap.gif',
+            },
+        ]
+        self.asset.deployment.mock_submissions(self.submissions)
+
+        data = {
+            'query': f'{{"{group}":{{"$elemMatch":{{"{group}/{question}":{{"$exists":true}}}}}}}}',
+            'format': 'json',
+        }
+        response = self.client.get(self.submission_list_url, data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        if isinstance(response.data, list):
```

Comment:
This is pretty bad but the alternative seems to be writing two 95% identical copies of the test for each api version which I think is worse.